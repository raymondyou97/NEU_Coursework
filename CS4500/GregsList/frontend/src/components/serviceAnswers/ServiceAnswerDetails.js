import React from 'react';
import '../../styles/serviceQADetails.css';
class ServiceAnswerDetails extends React.Component {
  componentDidMount() {
    this.props.fetchServiceAnswer();
  }

  render() {
    return (
      <div>
        <h3>Service Answer Details</h3>
        <div className="service-q-a-details-whole-block">
          <div className="service-q-a-details-field">
            <span className="service-q-a-details-field-header">ID</span>
            <select
              value={this.props.id}
              onChange={e => this.props.selectServiceAnswer(e.target.value)}
              className="form-control service-q-a-details-id-select"
            >
              {this.props.serviceAnswers.map(serviceAnswer => (
                <option value={serviceAnswer.id} key={serviceAnswer.id}>
                  {serviceAnswer.id}
                </option>
              ))}
            </select>
          </div>
          <div className="service-q-a-details-field">
            <span className="service-q-a-details-field-header">True/False</span>
            <input
              onChange={this.props.handleTrueFalseChange}
              className="form-control service-q-a-details-field-input"
              value={this.props.serviceAnswer.trueFalseAnswer}
            />
          </div>
          <div className="service-q-a-details-field">
            <span className="service-q-a-details-field-header">
              Min Range Answer
            </span>
            <input
              onChange={this.props.handleMinRangeAnswer}
              className="form-control service-q-a-details-field-input"
              value={this.props.serviceAnswer.minRangeAnswer}
            />
          </div>
          <div className="service-q-a-details-field">
            <span className="service-q-a-details-field-header">
              Max Range Answer
            </span>
            <input
              onChange={this.props.handleMaxRangeAnswer}
              className="form-control service-q-a-details-field-input"
              value={this.props.serviceAnswer.maxRangeAnswer}
            />
          </div>
          <div className="service-q-a-details-field">
            <span className="service-q-a-details-field-header">
              Choice Answer
            </span>
            <input
              onChange={this.props.handleChoiceAnswer}
              className="form-control service-q-a-details-field-input"
              value={this.props.serviceAnswer.choiceAnswer}
            />
          </div>
          <div>
            <button
              className="service-q-a-details-button"
              onClick={this.props.backToAnswers}
            >
              Cancel
            </button>
            <button
              className="service-q-a-details-button"
              onClick={() =>
                this.props.deleteServiceAnswer(this.props.serviceAnswer.id)
              }
            >
              Delete
            </button>
            <button
              className="service-q-a-details-button"
              onClick={this.props.createServiceAnswer}
            >
              Create
            </button>
            <button
              className="service-q-a-details-button"
              onClick={this.props.updateServiceAnswer}
            >
              Update
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default ServiceAnswerDetails;
