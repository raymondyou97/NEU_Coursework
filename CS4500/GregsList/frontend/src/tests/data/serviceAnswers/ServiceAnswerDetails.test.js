'use es6';

import React from 'react';
import { shallow } from 'enzyme';
import ServiceAnswerDetails from '../../../components/serviceAnswers/ServiceAnswerDetails';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SERVICE_ANSWER_LIST } from './ServiceAnswerData';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

test('Snapshot renders with data for selected answer', () => {
  let selectedServiceAnswer = SERVICE_ANSWER_LIST[2];

  const component = renderer.create(
    <ServiceAnswerDetails
      id={selectedServiceAnswer.id}
      serviceAnswers={SERVICE_ANSWER_LIST}
      serviceAnswer={selectedServiceAnswer}
      fetchServiceAnswer={jest.fn()}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot renders with null data for selected answer', () => {
  let selectedServiceAnswer = {
    id: null,
    trueFalseAnswer: null,
    minRangeAnswer: null,
    maxRangeAnswer: null,
    choiceAnswer: null,
  };

  const component = renderer.create(
    <ServiceAnswerDetails
      id={null}
      serviceAnswers={SERVICE_ANSWER_LIST}
      serviceAnswer={selectedServiceAnswer}
      fetchServiceAnswer={jest.fn()}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Initial DOM contains all fields with proper values for selected answer', () => {
  let selectedServiceAnswer = SERVICE_ANSWER_LIST[2];

  let component = shallow(
    <ServiceAnswerDetails
      id={selectedServiceAnswer.id}
      serviceAnswers={[]}
      serviceAnswer={selectedServiceAnswer}
      fetchServiceAnswer={jest.fn()}
    />
  );

  let selectedId = component.find('.service-q-a-details-id-select');
  let inputFields = component.find('.service-q-a-details-field-input');

  expect(selectedId.props().value).toEqual(3);
  expect(inputFields.length).toEqual(4);
  expect(inputFields.at(0).props().value).toEqual(true);
  expect(inputFields.at(1).props().value).toEqual(50);
  expect(inputFields.at(2).props().value).toEqual(100);
  expect(inputFields.at(3).props().value).toEqual(51);
});

test('Initial DOM contains all fields with null values for empty selected answer', () => {
  let selectedServiceAnswer = {
    id: null,
    trueFalseAnswer: null,
    minRangeAnswer: null,
    maxRangeAnswer: null,
    choiceAnswer: null,
  };

  let component = shallow(
    <ServiceAnswerDetails
      id={null}
      serviceAnswers={[]}
      serviceAnswer={selectedServiceAnswer}
      fetchServiceAnswer={jest.fn()}
    />
  );

  let selectedId = component.find('.service-q-a-details-id-select');
  let inputFields = component.find('.service-q-a-details-field-input');

  expect(selectedId.props().value).toEqual(null);
  expect(inputFields.length).toEqual(4);
  expect(inputFields.at(0).props().value).toEqual(null);
  expect(inputFields.at(1).props().value).toEqual(null);
  expect(inputFields.at(2).props().value).toEqual(null);
  expect(inputFields.at(3).props().value).toEqual(null);
});
