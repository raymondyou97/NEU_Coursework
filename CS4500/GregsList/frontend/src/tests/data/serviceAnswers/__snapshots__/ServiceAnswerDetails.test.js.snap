// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders with data for selected answer 1`] = `
<div>
  <h3>
    Service Answer Details
  </h3>
  <div
    className="service-q-a-details-whole-block"
  >
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        ID
      </span>
      <select
        className="form-control service-q-a-details-id-select"
        onChange={[Function]}
        value={3}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
      </select>
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        True/False
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={true}
      />
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        Min Range Answer
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={50}
      />
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        Max Range Answer
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={100}
      />
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        Choice Answer
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={51}
      />
    </div>
    <div>
      <button
        className="service-q-a-details-button"
      >
        Cancel
      </button>
      <button
        className="service-q-a-details-button"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="service-q-a-details-button"
      >
        Create
      </button>
      <button
        className="service-q-a-details-button"
      >
        Update
      </button>
    </div>
  </div>
</div>
`;

exports[`Snapshot renders with null data for selected answer 1`] = `
<div>
  <h3>
    Service Answer Details
  </h3>
  <div
    className="service-q-a-details-whole-block"
  >
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        ID
      </span>
      <select
        className="form-control service-q-a-details-id-select"
        onChange={[Function]}
        value={null}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
      </select>
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        True/False
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={null}
      />
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        Min Range Answer
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={null}
      />
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        Max Range Answer
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={null}
      />
    </div>
    <div
      className="service-q-a-details-field"
    >
      <span
        className="service-q-a-details-field-header"
      >
        Choice Answer
      </span>
      <input
        className="form-control service-q-a-details-field-input"
        value={null}
      />
    </div>
    <div>
      <button
        className="service-q-a-details-button"
      >
        Cancel
      </button>
      <button
        className="service-q-a-details-button"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="service-q-a-details-button"
      >
        Create
      </button>
      <button
        className="service-q-a-details-button"
      >
        Update
      </button>
    </div>
  </div>
</div>
`;
